File: FixMyCar/FixMyCar.UI/fixmycar_car_parts_shop/lib/src/models/store_item/store_item.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:fixmycar_car_parts_shop/src/models/car_model/car_model.dart';

part 'store_item.g.dart';

@JsonSerializable()
class StoreItem {
  int id;
  String name;
  double price;
  String state;
  double discount;
  double discountedPrice;
  String? imageData;
  String? details;
  List<CarModel>? carModels;
  int? storeItemCategoryId;
  String? category;

  StoreItem(
      this.id, this.name, this.price, this.state, this.discount, this.discountedPrice, this.imageData, this.details, this.carModels, this.storeItemCategoryId, this.category);

  factory StoreItem.fromJson(Map<String, dynamic> json) => _$StoreItemFromJson(json);

  Map<String, dynamic> toJson() => _$StoreItemToJson(this);
}

File: FixMyCar/FixMyCar.Model/Entities/StoreItem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FixMyCar.Model.Entities
{
    public partial class StoreItem
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public double Price { get; set; }
        public double Discount { get; set; }
        public double DiscountedPrice { get; set; }
        public string State { get; set; }
        public byte[]? ImageData { get; set; }
        public string? Details { get; set; }
        [ForeignKey(nameof(CarPartsShop))]
        public int CarPartsShopId { get; set; }
        public CarPartsShop CarPartsShop { get; set; }
        public ICollection<StoreItemCarModel>? StoreItemCarModels { get; set; }
        public ICollection<CarModel>? CarModels { get; set; }
        [ForeignKey(nameof(StoreItemCategory))]
        public int? StoreItemCategoryId { get; set; }
        public StoreItemCategory? StoreItemCategory { get; set; }
        public ICollection<OrderDetail> OrderDetails { get; set; }
    }
}